### PR TITLE
Make compatible with Python 3.8 on an M1 MacBook

### PR DESCRIPTION
**Pull request by GraemeWatt**
This PR makes a few changes to requirements and installation documentation to enable the HEPData software to be installed on a MacBook with an M1 processor and using Python 3.8.  The CI is extended to test using Python 3.7 and Python 3.8 in addition to Python 3.6 (currently used in production).  End-to-end tests are only run for Python 3.6 as they seem to fail if running multiple jobs in parallel.  Before merging, let's see if @zeynepaki can install successfully and if she has any feedback on the installation documentation.

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -82,6 +82,7 @@ To run outside of Docker you can use the Homebrew installation of Elasticsearch
 
     $ brew install elasticsearch
 
+Neither of these two methods is currently working for an M1 MacBook.
```
Comment (alisonrclarke) [inline INSTALL.rst]: So what are you doing currently?

File: INSTALL.rst [hunk 2]
```diff
@@ -91,19 +92,33 @@ Installation
 Python
 ------
 The HEPData code is only compatible with Python 3 (not Python 2).  It has been tested with Python 3.6.
+It has also been tested with Python 3.8 on an M1 MacBook where some changes were required (documented below).
 
-First install all requirements in a `virtualenv <https://virtualenv.pypa.io/en/stable/installation.html>`_
-using `virtualenvwrapper <https://virtualenvwrapper.readthedocs.io/en/latest/install.html>`_:
+First install all requirements in a `virtualenv <https://virtualenv.pypa.io/en/stable/installation.html>`_.
+(Use `virtualenvwrapper <https://virtualenvwrapper.readthedocs.io/en/latest/install.html>`_ if you prefer.)
```
Comment (alisonrclarke) [inline INSTALL.rst:98]: Could you clarify that the instructions below are for using virtualenv directly, not venvwrapper?

File: INSTALL.rst [hunk 3]
```diff
@@ -136,6 +151,11 @@ Next, build assets using webpack (via `invenio-assets <https://invenio-assets.re
 
    (hepdata)$ ./scripts/clean_assets.sh
 
+On an M1 MacBook, until an `issue with Invenio-Assets <https://github.com/inveniosoftware/invenio-assets/issues/144>`_
+is addressed, you will need to replace
+``"node-sass": "^4.12.0",`` with ``"sass": "^1.50.0",`` (or another `Dart Sass <https://sass-lang.com/dart-sass>`_
+version) in the ``package.json`` file of the ``invenio-assets`` installation.
```
Comment (alisonrclarke) [inline INSTALL.rst]: Could you clarify for new devs where to find the package.json file?

File: INSTALL.rst [hunk 4]
```diff
@@ -149,20 +169,24 @@ PostgreSQL
...
+   $ createdb hepdata -O hepdata
+   $ createdb hepdata_test -O hepdata
 
 Next, create the database and database tables.
 Also create a user and populate the database with some records.
+Make sure that Celery is running before proceeding further.
+Until an `issue <https://github.com/HEPData/hepdata/issues/461>`_ is addressed and ``Invenio-Accounts`` is upgraded
+to at least v1.4.9, you will need to manually
+`patch <https://github.com/inveniosoftware/invenio-accounts/commit/b91649244b11479d8fa817745141c0027001dff1>`_
+the ``invenio_accounts/cli.py`` file before the next step.
```
Comment (alisonrclarke) [inline INSTALL.rst]: Again, could you clarify the path (if the user has used venv as above)?

File: INSTALL.rst [hunk 5]
```diff
@@ -175,18 +199,18 @@ Inspect the ``hepdata`` database from the command line as the ``hepdata`` user:
 
    $ psql hepdata -U hepdata -h localhost
    Password for user hepdata: hepdata
-   hepdata=> select publication_recid, inspire_id, last_updated from hepsubmission;
+   hepdata=> select publication_recid, inspire_id, last_updated from hepsubmission order by publication_recid;
 
+   hepdata=> select publication_recid, inspire_id, last_updated from hepsubmission order by publication_recid;
```
Comment (alisonrclarke) [inline INSTALL.rst:206]: Is this line in twice?

File: INSTALL.rst (hunk 1 above)
Comment (GraemeWatt) [reply INSTALL.rst]: For now I'm just using Elasticsearch v7.1.1 installed with Homebrew.  I added a clarifying comment.  I guess we'll need to move to OpenSearch fairly soon, which might be easier to install on an M1 MacBook.

File: INSTALL.rst (hunk 2 above)
Comment (GraemeWatt) [reply INSTALL.rst:98]: OK, done.

File: INSTALL.rst (hunk 3 above)
Comment (GraemeWatt) [reply INSTALL.rst]: Good point, it's not obvious.  I added an explanation.

File: INSTALL.rst (hunk 4 above)
Comment (GraemeWatt) [reply INSTALL.rst]: Yes, done.

File: INSTALL.rst (hunk 5 above)
Comment (GraemeWatt) [reply INSTALL.rst:206]: Thanks for spotting.  I removed the duplicate line.

**Review by alisonrclarke — APPROVED** [review]

**Pull request merged**